ensime_shared/protocol.py

```diff
@@ -138,16 +131,18 @@ def add(member, indentLevel):
             indent = "  " * indentLevel
             t = member["declAs"]["typehint"] if member["typehint"] == "BasicTypeInfo" else ""
             line = "{}{}: {}".format(indent, t, member["name"])
-            self.vim.command(commands["append_line"].format("\'$\'", str(line)))
+            self.editor.append(line)
             if indentLevel < 4:
                 for m in member["members"]:
                     add(m, indentLevel + 1)
 
         # Create a new buffer 45 columns wide
-        cmd = commands["new_vertical_scratch"].format(str(45), "package_info")
-        self.vim.command(cmd)
-        self.vim.command(commands["set_filetype"].format("package_info"))
-        self.vim.command(commands["append_line"].format("\'$\'", str(package)))
+        opts = {'buftype': 'nofile', 'bufhidden': 'wipe',
+                'buflisted': False, 'swapfile': False}
+        self.editor.split_window('package_info', vertical=True, size=45, bufopts=opts)
+        self.editor.set_filetype('package_info')
+
+        self.editor.append(str(package))
```

Comment:
It'd be nice to have an even higher-level API like `Editor.open_package_inspector()` or something like that so that message handler code could think in terms of "UI components" and not low-level Vim buffer options, but that's the sort of stuff I want to save for a later review/design session.